src/platform/backends/shared/linux/process_factory.cpp

```diff
@@ -64,7 +64,7 @@ class AppArmoredProcess : public mp::BasicProcess
 
 mp::optional<mp::AppArmor> create_apparmor()
 {
-    if (!mp::utils::is_snap() && qEnvironmentVariableIsSet("DISABLE_APPARMOR"))
+    if (!qEnvironmentVariableIsSet("SNAP") && qEnvironmentVariableIsSet("DISABLE_APPARMOR"))
```

Comment:
Why a different approach here? I mean, it could also be something like:

```c++
try
{
    mu::snap_dir();
    if (qEnvironmentVariableIsSet("DISABLE_APPARMOR")
    {
        ...
    }
}
catch(...)
...
```

Reply:
Ok, I agree with you that is a more consistent way of handling this.  Thanks!

Reply:
I redid this logic taking the exception into account now.